### ISSUE TITLE
Segfault

### ISSUE THREAD

**Issue by DalenW** [arthurlm/one-brc-rs#1]
Here's what the debugger says:
![Screenshot 2024-02-04 at 12 44 14 PM](https://github.com/arthurlm/one-brc-rs/assets/25020780/88c64126-3a3d-430e-bb69-1ef5217c9fb1)

**Comment by arthurlm** [comment]
Looking at the locals variables from your screen capture here is my guess on what happen:

1. `i2 == 110`
2. `i2 / 10 == 11`
3. `i2 / 10 == b';' as u8 & 0x0F`
4. `i2 / 10 == line[line.len() - 3]`
5. `line.len() == 5`
6. `line[2] = b';'`
7. `line[4] = b'6'` (because `f=6`)

So my guess is the input file you provide contains line with following format:
```txt
??;?6
```

As you can see, the parser I wrote rely a lot on `unsafe` code to strip out all check from the final x86 assembly. 
Lines produced by the 1brc measures generator **must** respect the following regexp format:

```txt
\w+;-?\d{1,2}\.\d
```

Did you use the `./create_measurements.sh` script provided in the official repository ? Or did you use another script to generate the input file ?

**Comment by DalenW** [comment]
Yeah I did use a different script to generate the file. My bad. I'll try to rerun it with the script you mentioned when I find a time to do so.
